client/asset/eth/eth.go

```diff
@@ -131,42 +131,29 @@ func (d *Driver) Create(params *asset.CreateWalletParams) error {
 	return CreateWallet(params)
 }
 
-// rawWallet is an unexported return type from the eth client. Watch for changes at
-// https://github.com/ethereum/go-ethereum/blob/c503f98f6d5e80e079c1d8a3601d188af2a899da/internal/ethapi/api.go#L227-L253
-type rawWallet struct {
-	URL      string             `json:"url"`
-	Status   string             `json:"status"`
-	Failure  string             `json:"failure,omitempty"`
-	Accounts []accounts.Account `json:"accounts,omitempty"`
-}
-
 // ethFetcher represents a blockchain information fetcher. In practice, it is
 // satisfied by rpcclient. For testing, it can be satisfied by a stub.
 type ethFetcher interface {
-	accounts() []*accounts.Account
-	addPeer(ctx context.Context, peer string) error
-	balance(ctx context.Context, addr *common.Address) (*big.Int, error)
+	address() common.Address
+	balance(ctx context.Context) (*big.Int, error)
 	bestBlockHash(ctx context.Context) (common.Hash, error)
 	bestHeader(ctx context.Context) (*types.Header, error)
 	block(ctx context.Context, hash common.Hash) (*types.Block, error)
 	blockNumber(ctx context.Context) (uint64, error)
-	connect(ctx context.Context, node *node.Node, contractAddr *common.Address) error
-	listWallets(ctx context.Context) ([]rawWallet, error)
-	initiate(txOpts *bind.TransactOpts, netID int64, initiations []dexeth.ETHSwapInitiation) (*types.Transaction, error)
-	estimateGas(ctx context.Context, msg ethereum.CallMsg) (uint64, error)
-	lock(ctx context.Context, acct *accounts.Account) error
-	nodeInfo(ctx context.Context) (*p2p.NodeInfo, error)
-	pendingTransactions(ctx context.Context) ([]*types.Transaction, error)
-	peers(ctx context.Context) ([]*p2p.PeerInfo, error)
-	transactionReceipt(ctx context.Context, txHash common.Hash) (*types.Receipt, error)
-	sendTransaction(ctx context.Context, tx map[string]string) (common.Hash, error)
+	connect(ctx context.Context) error
+	initiate(ctx context.Context, contracts []*asset.Contract, maxFeeRate uint64, contractVer uint32) (*types.Transaction, error)
```

Comment:
The `contractVer` doesn't need to map precisely to the `dex.Asset.Version`, but as long as we make a rule that a change in contract always accompanies a bump in `dex.Asset.Version`, then we can find a scheme to get the contract version from the asset version and we don't need any new data.